### PR TITLE
Install jira plugins

### PR DESCRIPTION
**Pull request by mbrehovsky**
Suggestions
- maybe add wait for jira ready (reuse the plugin_installation_completed)
- add support for other applications

### CODE REVIEW

File: bin/avst-app [hunk 1]
```diff
@@ -141,6 +141,9 @@ else
                     ;;
                 restore)
                     ;;
+                install_plugins)
+                    warn "${THIS_COMMAND}: Application is installed but not yet modified, would be nice to run modify first. Proceedon your own risk."
```
Comment (jonmort) [inline bin/avst-app]: Proceed at your own risk

File: share/avst-app/lib/product/jira/install_plugins.d/05install_plugins [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# Copyright 2015 Adaptavist.com Ltd.
```
Comment (jonmort) [inline share/avst-app/lib/product/jira/install_plugins.d/05install_plugins]: update copyright date?

File: share/avst-app/lib/common/helpers.sh [hunk 3]
```diff
@@ -433,4 +433,120 @@ function signal_stop_process () {
...
+    MARKETPLACE_URL="https://marketplace.atlassian.com"
+    MARKETPLACE_ADDONS_URL="${MARKETPLACE_URL}/rest/2/addons"
+
+    if [[ "${PLUGIN_VERSION}" == "latest" ]]; then
+        URL="${MARKETPLACE_ADDONS_URL}/${PLUGIN_KEY}/versions/latest"
+    else
+        URL="${MARKETPLACE_ADDONS_URL}/${PLUGIN_KEY}/versions/name/${PLUGIN_VERSION}"
+    fi
+    
+    RESP=$(curl -s -H "Accept: application/json" \
```
Comment (jonmort) [inline share/avst-app/lib/common/helpers.sh]: Should we be checking the response code? Would need to pipe the output to a file i guess, but that would mean that the sed command would not need the echo

File: share/avst-app/lib/common/helpers.sh [hunk 4]
```diff
@@ -433,4 +433,120 @@ function signal_stop_process () {
...
 
- 
+# Retrieves access token to running jira instance
+function get_application_access_token () {
+    APPLICATION_URL="${1}"
+    APPLICATION_ADMIN_USER="${2}"
+    APPLICATION_ADMIN_PASS="${3}"
+    URL="${APPLICATION_URL}/rest/plugins/1.0/?os_authType=basic"
+
+    shopt -s extglob # Required to trim whitespace; see below
```
Comment (jonmort) [inline share/avst-app/lib/common/helpers.sh]: Should this be reset after we are done?

File: share/avst-app/lib/common/helpers.sh [hunk 5]
```diff
@@ -433,4 +433,128 @@ function signal_stop_process () {
     fi
 }
 
- 
+# Retrieves access token to running jira instance
+function get_application_access_token () {
+    APPLICATION_URL="${1}"
+    APPLICATION_ADMIN_USER="${2}"
+    APPLICATION_ADMIN_PASS="${3}"
+    URL="${APPLICATION_URL}/rest/plugins/1.0/?os_authType=basic"
```
Comment (mattohope) [inline share/avst-app/lib/common/helpers.sh]: could we have "/rest/plugins/1.0" in a variable defaulted to that value, this allows us to change it but also its used many times in this file

File: share/avst-app/lib/common/helpers.sh [hunk 6]
```diff
@@ -433,4 +433,128 @@ function signal_stop_process () {
...
+}
+
+function install_plugin () {
+    PLUGIN_KEY="${1}"
+    PLUGIN_BUILD_NUMBER="${2}"
+    TOKEN="${3}"
+    APPLICATION_URL="${4}"
+    APPLICATION_ADMIN_USER="${5}"
+    APPLICATION_ADMIN_PASS="${6}"
+    URL="${APPLICATION_URL}/rest/plugins/1.0/?token=${TOKEN}"
```
Comment (mattohope) [inline share/avst-app/lib/common/helpers.sh]: could we have "/rest/plugins/1.0" in a variable defaulted to that value, this allows us to change it but also its used many times in this file

File: share/avst-app/lib/common/helpers.sh [hunk 7]
```diff
@@ -433,4 +433,128 @@ function signal_stop_process () {
...
+    done
+}
+
+function install_plugin_license () {
+    PLUGIN_KEY="${1}"
+    PLUGIN_LICENSE="${2}"
+    APPLICATION_URL="${3}"
+    APPLICATION_ADMIN_USER="${4}"
+    APPLICATION_ADMIN_PASS="${5}"
+    URL="${APPLICATION_URL}/rest/plugins/1.0/${PLUGIN_KEY}-key/license"
```
Comment (mattohope) [inline share/avst-app/lib/common/helpers.sh]: could we have "/rest/plugins/1.0" in a variable defaulted to that value, this allows us to change it but also its used many times in this file

**Pull request merged**